client/client.go

```diff
@@ -169,25 +155,57 @@ func (c *DatabricksClient) attempt(ctx context.Context,
 		request.Header.Set("User-Agent", useragent.FromContext(request.Context()))
 
 		// attempt the actual request
-		resp, err := c.httpClient.Do(request)
-		retry, err := apierr.CheckForRetry(ctx, resp, err)
+		response, err := c.httpClient.Do(request)
+
+		// Read responseBody immediately because we need it regardless of success or failure.
+		//
+		// Fully read and close HTTP response responseBody to release HTTP connections.
+		//
+		// HTTP client's Transport may not reuse HTTP/1.x "keep-alive" TCP connections
+		// if the Body is not read to completion and closed.
+		//
+		// See: https://groups.google.com/g/golang-nuts/c/IoSvPz-rpfc
+		var responseBody bytes.Buffer
+		var responseBodyErr error
+		if response != nil {
+			_, responseBodyErr = responseBody.ReadFrom(response.Body)
+			response.Body.Close()
+		}
+
+		retry, err := apierr.CheckForRetry(ctx, response, err, responseBody.Bytes(), responseBodyErr)
+		if err != nil && !errors.As(err, &apierr.APIError{}) {
+			err = fmt.Errorf("failed request: %w", err)
+		}
+		if err == nil && response == nil {
+			err = fmt.Errorf("no response: %s %s", method, requestURL)
```

Comment:
this _technically_ should not happen because default http client always has to return a response, but it's good to check, yes

Reply:
Copied from the other code to make all tests pass ;-)